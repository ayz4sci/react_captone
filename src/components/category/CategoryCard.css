h5 {
    background: aliceblue;
}

@media (min-width: 62em) {
    .card-columns {
        -webkit-column-count: 4!important;
        -moz-column-count: 4!important;
        column-count: 4!important;
    }
}