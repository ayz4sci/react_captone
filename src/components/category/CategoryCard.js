import React from 'react';
import PropTypes from 'prop-types';
import Cheat from '../cheat/Cheat';
import './CategoryCard.css'

class CategoryCard extends React.Component{
    render(){
        return (
            <div className="card m-1">
                <h5 className="p-2 text-capitalize">{this.props.category.category}</h5>
                { this.props.category.cheats.map( (cheat, key) => 
                    <Cheat cheat={cheat} key={key} />
                )}
            </div>
        )
    }
}

CategoryCard.protoTypes = {
    category: PropTypes.shape({
        name: PropTypes.string.isRequired,
        cheats: PropTypes.array.isRequired 
    }).isRequired
}

export default CategoryCard;