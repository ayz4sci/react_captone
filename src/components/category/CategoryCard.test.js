import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import CategoryCard from './CategoryCard';

it('renders correctly', () => {
    const props = { 
        category: {
            name: "lol",
            cheats: []
        }
    };

    const wrapper = shallow(<CategoryCard {...props} />, {mode: 'shallow'})
    expect(toJson(wrapper)).toMatchSnapshot();
});