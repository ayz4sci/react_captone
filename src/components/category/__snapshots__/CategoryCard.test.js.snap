// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="card m-1"
>
  <h5
    className="p-2 text-capitalize"
  >
    lol
  </h5>
</div>
`;
